File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapNodes(ListNode* head, int k) {
        ListNode *first=head;
        ListNode *slow=head;
        ListNode *kth=nullptr;
        while(--k){
            first=first->next;
        }
        kth=first;
        first=first->next;
        while(first){
            slow=slow->next;
            first=first->next;
        }
        swap(slow->val,kth->val);
        return head;
    }
};

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        first = last = head
        for i in range(1,k):
            first = first.next
            
        nullchecker = first
        while nullchecker.next:
            nullchecker=nullchecker.next
            last=last.next
        first.val,last.val=last.val,first.val
        return head